src/compiler/c.rs

```diff
@@ -521,20 +521,21 @@ impl pkg::InputsPackager for CInputsPackager {
         let mut builder = tar::Builder::new(wtr);
 
         {
-            let input_path = pkg::simplify_path(&input_path)?;
+            let (input_path, dirs) = pkg::simplify_path(&input_path)?;
+            assert!(dirs.is_empty()); // must be simplified already
             let dist_input_path = path_transformer.as_dist(&input_path).with_context(|| {
                 format!("unable to transform input path {}", input_path.display())
             })?;
 
-            let mut file_header = pkg::make_tar_header(&input_path, &dist_input_path)?;
+            let mut file_header = pkg::make_tar_header(&input_path, &dist_input_path, false)?;
             file_header.set_size(preprocessed_input.len() as u64); // The metadata is from non-preprocessed
             file_header.set_cksum();
             builder.append(&file_header, preprocessed_input.as_slice())?;
         }
 
         for input_path in extra_hash_files {
-            let input_path = pkg::simplify_path(&input_path)?;
-
+            let (input_path, dirs) = pkg::simplify_path(&input_path)?;
+            assert!(dirs.is_empty()); // must be simplified already
```

Comment:
This `assert` is a potential footgun, a warning or `let (a,_) = ..` would be more appropriate.

Reply:
in case "dirs" is non-empty here, the compiler invocation will fail (with sccache-dist) due to these directories not existing.

this is in implementation of InputsPackager, which is only run if "dist-client" is enabled.

given the inevitable failure, the assertion failure is preferable to a more easily ignored warning.

the intention (and i hope current situation in this PR) is that the simplify_path is already done by code that runs previously (in "gcc.rs"), and so i have removed the code that would handle a non-empty "dirs" in "c.rs".

this was to implement the idea by mitchhentges in this previous comment https://github.com/mozilla/sccache/pull/1126#pullrequestreview-887421393

there is another such assert in "rust.rs" - there the situation is different, in that there is no earlier call to simplify_path, but it is the same in that the compiler invocation will fail because the directory doesn't exist - already the case before this PR.